cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -306,18 +308,62 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
+  liftIO $ LBS.writeFile (outputDir </> "alonzo-genesis.json") $ Aeson.encode alonzoGenesis
   where
     genesisDir = outputDir </> "genesis-keys"
     delegateDir = outputDir </> "delegate-keys"
     drepsDir = outputDir </> "drep-keys"
     utxoKeysDir = outputDir </> "utxo-keys"
     poolsDir = outputDir </> "pools-keys"
     stakeDelegatorsDir = outputDir </> "stake-delegators"
-    numStakeDelegators = case stakeDelegators of OnDisk n -> n; Transient n -> n
     mkDelegationMapEntry :: Delegation -> (L.KeyHash L.Staking L.StandardCrypto, L.PoolParams L.StandardCrypto)
     mkDelegationMapEntry d = (dDelegStaking d, dPoolParams d)
 
+    addDRepsToConwayGenesis :: [VerificationKey DRepKey] -> [VerificationKey StakeKey]
+                            -> L.ConwayGenesis L.StandardCrypto -> L.ConwayGenesis L.StandardCrypto
+    addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
+      conwayGenesis { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+                    , L.cgInitialDReps = initialDReps (L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis) dRepKeys
+                    }
+
+      where
+      delegs :: [(VerificationKey StakeKey, VerificationKey DRepKey)] -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      delegs = ListMap.fromList . map (bimap verificationKeytoStakeCredential (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential))
+
+      initialDReps :: L.Coin -> [VerificationKey DRepKey] -> ListMap (L.Credential L.DRepRole L.StandardCrypto) (L.DRepState L.StandardCrypto)
+      initialDReps minDeposit = ListMap.fromList . map (\c -> ( verificationKeyToDRepCredential c
+                                                              , L.DRepState { L.drepExpiry = EpochNo 1000
+                                                                            , L.drepAnchor = SNothing
+                                                                            , L.drepDeposit = max (L.Coin 1_000_000) minDeposit
+                                                                            }))
+
+      verificationKeyToDRepCredential :: VerificationKey DRepKey -> L.Credential L.DRepRole L.StandardCrypto
+      verificationKeyToDRepCredential vk = dRepKeyToCredential (verificationKeyHash vk)
+        where
+          dRepKeyToCredential :: Hash DRepKey -> L.Credential L.DRepRole L.StandardCrypto
+          dRepKeyToCredential (DRepKeyHash v) = L.KeyHashObj v
+
+      verificationKeytoStakeCredential :: VerificationKey StakeKey -> L.Credential L.Staking L.StandardCrypto
+      verificationKeytoStakeCredential vk = stakeKeyToCredential (verificationKeyHash vk)
+        where
+          stakeKeyToCredential :: Hash StakeKey -> L.Credential L.Staking L.StandardCrypto
+          stakeKeyToCredential (StakeKeyHash v) = L.KeyHashObj v
+
+    zipWithDeepSeq :: NFData c => (a -> b -> c) -> [a] -> [b] -> [c]
```

Comment:
A comment here regarding why we do this would be useful

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/682